moveit_core/robot_model/src/joint_model_group.cpp

```diff
@@ -831,5 +831,41 @@ bool JointModelGroup::isValidVelocityMove(const double* from_joint_pose, const d
 
   return true;
 }
+
+std::pair<Eigen::VectorXd, Eigen::VectorXd> getLowerAndUpperLimits(const moveit::core::JointModelGroup& group)
+{
+  // Get the group joints lower/upper position limits.
+  Eigen::VectorXd lower_limits(group.getActiveVariableCount());
+  Eigen::VectorXd upper_limits(group.getActiveVariableCount());
+  const moveit::core::JointBoundsVector& group_bounds = group.getActiveJointModelsBounds();
+  int joint_index = 0;
+  for (const moveit::core::JointModel::Bounds* joint_bounds : group_bounds)
+  {
+    for (const moveit::core::VariableBounds& variable_bounds : *joint_bounds)
+    {
+      lower_limits[joint_index] = variable_bounds.min_position_;
+      upper_limits[joint_index] = variable_bounds.max_position_;
+      joint_index++;
+    }
+  }
+  return { lower_limits, upper_limits };
+}
+
+std::pair<Eigen::VectorXd, Eigen::VectorXd>
+getMaximumVelocitiesAndAccelerations(const moveit::core::JointModelGroup& group)
+{
+  const std::size_t n_dofs = group.getActiveVariableCount();
+  Eigen::VectorXd max_joint_velocities = Eigen::VectorXd::Constant(n_dofs, 0.0);
+  Eigen::VectorXd max_joint_accelerations = Eigen::VectorXd::Constant(n_dofs, 0.0);
+  const moveit::core::JointBoundsVector& bounds = group.getActiveJointModelsBounds();
+  assert(bounds.size() == n_dofs);
+  for (std::size_t i = 0; i < bounds.size(); ++i)
+  {
+    assert(bounds[i]->size() == 1);  // single-variable joints.
+    max_joint_velocities[i] = std::min(-bounds[i]->at(0).min_velocity_, bounds[i]->at(0).max_velocity_);
```

Comment:
If max. velocities or accelerations are not symmetric (e.g. [-1, 2]), this function will return the most limiting element (i.e. 1). Maybe worth clarifying that in the doc in the header, and maybe rename the function to `getMostLimitingSymmetricVelocitiesAndAccelerations`, which I realize it's probably too long...

Reply:
I named it getMaxVelocitiesAndAccelerationBounds what do you think about that?